net/libfab/libfab.c

```diff
@@ -950,23 +969,12 @@ static void libfab_poller(struct m0_fab__tm *tm)
 
 	libfab_tm_event_post(tm, M0_NET_TM_STARTED);
 	while (tm->ftm_state != FAB_TM_SHUTDOWN) {
-		/*
-		 * The lab team observed a significant increase in the CPU load
-		 * due to the epoll_wait not sleeping.
-		 * The proposed short term fix to support the lab team is adding
-		 * nanosleep of 0ns on VM to reduce CPU load.
-		 */
-		if (m0_processor_is_vm())
-			m0_nanosleep(M0_MKTIME(0 ,0), NULL);
-		/*
-		 * It is observed that with epoll_wait,
-		 * the thread is waiting in a busy-loop for events
-		 * thus not releasing CPU.
-		 * Hence, adding a sched_yield() will release the CPU for
-		 * other processes and reduce CPU consumption.
-		 */
-		sched_yield();
-		ev_cnt = epoll_wait(tm->ftm_epfd, &ev, 1, FAB_WAIT_FD_TMOUT);
+		if (fi_trywait(tm->ftm_fab->fab_fab, tm->ftm_fids.ftf_head,
```

Comment:
Suggestion: we need to check the return codes of fi_trywait and epoll_wait, we need to go back to waiting in case of EINTR. For example,

```c
...
rc = fi_trywait(...);
M0_ASSERT(M0_IN(rc, FI_SUCCESS, -FI_EAGAIN));
if (rc == FI_SUCCESS) {
  rc = -EINTR;
  while (rc == -EINTR) {
    rc = epoll_wait(...);
    M0_ASSERT(M0_IN(rc, (-EINTR, 0, 1));
  }
  ev_cnt = rc;
} else
  ev_cnt = 0;
...
```

Reply:
Added handling of -EINTR for epoll_wait().